src/AWS/S3/S3.purs

```diff
@@ -0,0 +1,130 @@
+module AWS.S3
+  ( S3
+  , BucketName
+  , BucketPolicyParams
+  , BucketPolicy
+  , createBucket
+  , CreateBucketResponse
+  , getObject
+  , GetObjectParams
+  , GetObjectResponse
+  , makeClient
+  , putBucketPolicy
+  ) where
+
+import AWS.Core.Client (makeClientHelper)
+import AWS.Core.Types (DefaultClientProps, Region(..))
+import Control.Promise (Promise, toAffE)
+import Data.Function.Uncurried (Fn2, runFn2, Fn3, runFn3)
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Foreign (Foreign)
+import Justifill (justifillVia)
+import Justifill.Fillable (class Fillable)
+import Justifill.Justifiable (class Justifiable)
+import Node.Buffer (Buffer)
+import Prelude (Unit, ($), class Show, (#), (<#>))
+import Type.Proxy (Proxy(..))
+import Data.Newtype (class Newtype)
+
+foreign import data S3 :: Type
+
+foreign import newS3 :: Foreign -> Effect S3
+
+makeClient ::
+  forall r via.
+  Justifiable { | r } { | via } =>
+  Fillable { | via } DefaultClientProps =>
+  { | r } ->
+  Effect S3
+makeClient r = makeClientHelper newS3 props
+  where
+  viaProxy :: Proxy { | via }
+  viaProxy = Proxy
+
+  props :: DefaultClientProps
+  props = justifillVia viaProxy r
+
+type InternalGetObjectParams
+  = { "Bucket" :: String
+    , "Key" :: String
+    }
+
+type InternalGetObjectResponse
+  = { "Body" :: Buffer
+    , "ContentLength" :: Int
+    , "ContentEncoding" :: String
+    , "ContentType" :: String
+    }
+
+foreign import getObjectImpl :: Fn2 S3 InternalGetObjectParams (Effect (Promise InternalGetObjectResponse))
+
+type GetObjectParams
+  = { bucket :: String
```

Comment:
Can we create newtypes for BucketKey? `bucket` should be the BucketName newtype, right?